modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -345,6 +350,10 @@ def _get_index(self):
         pandas.Index
             An index object containing the row labels.
         """
+        if self._index_cache is None:
+            self._index_cache, row_lengths = self._compute_axis_labels(0)
+            if self._row_lengths_cache is None:
```

Comment:
What if `row_lengths` gets changed?

Reply:
- This technically should have an `assert row_lengths == self._row_lengths_cache`, but I felt it would be too much of a performance penalty

- This should actually almost never happen that we have valid (non-`None`) row lengths but no index

Reply:
I can't think up a case when we hit this branch. Let's leave as is.